rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,198 @@
+    // strip the base reference url to get the full path that the file should be written to
+    // example: https://api.github.com/repos/org/repo/contents/kustomize.yml == kustomize.yml
+    return artifact.getReference().replace(artifact.getName(), "");
+  }
+
+  /**
+   * getFilesFromArtifact will use a single input artifact to determine the dependency tree of files
+   * mentoined it's (and subsequent) kustomization file.
+   */
+  private HashSet<String> getFilesFromArtifact(Artifact artifact) throws IOException {
+    // referenceBaseURL is the base URL without the artifacts name. works for github and bitbucket
+    // artifacts,
+    String referenceBaseURL = extractReferenceBase(artifact);
+    // filename is the base name of the file
+    String filename = FilenameUtils.getName(artifact.getName());
+    // get the base directory of the original file. we'll use this to fetch sibling files
+    Path base = Paths.get(artifact.getName()).getParent();
+    return getFilesFromArtifact(artifact, referenceBaseURL, base, filename);
+  }
+
+  private HashSet<String> getFilesFromArtifact(
+      Artifact artifact, String referenceBaseURL, Path base, String filename) throws IOException {
+    // filesToDownload will be used to collect all of the files we need to fetch later
+    HashSet<String> filesToDownload = new HashSet<>();
+
+    String referenceBase = referenceBaseURL.concat(base.toString());
+    Artifact testArtifact = artifactFromBase(artifact, referenceBase, base.toString());
+
+    Kustomization kustomization = kustomizationFileReader.getKustomization(testArtifact, filename);
+    // nonEvaluateFiles are files we know can't be references to other kustomizations
+    // so we know they only need to be collected for download later
+    Set<String> nonEvaluateFiles = kustomization.getFilesToDownload(referenceBase);
+    filesToDownload.addAll(nonEvaluateFiles);
+
+    for (String evaluate : kustomization.getFilesToEvaluate()) {
+      // we're assuming that files that look like folders are referencing
+      // kustomizations above or below the current directory. if the file doesn't
+      // look like a folder then we know it should be downloaded later.
+      if (isFolder(evaluate)) {
+        Path tmpBase = Paths.get(FilenameUtils.normalize(base.resolve(evaluate).toString()));
+        artifact.setName(tmpBase.toString());
+        filesToDownload.addAll(getFilesFromArtifact(artifact, referenceBaseURL, tmpBase, filename));
+      } else {
+        filesToDownload.add(referenceBase.concat(File.separator).concat(evaluate));
+      }
+    }
+    return filesToDownload;
+  }
+
+  private Artifact artifactFromBase(Artifact artifact, String reference, String name) {
+    return Artifact.builder()
+        .reference(reference)
+        .name(name)
+        .type(artifact.getType())
+        .artifactAccount(artifact.getArtifactAccount())
+        .build();
+  }
+
+  private boolean isFolder(String evaluate) {
```

Comment:
I'm not sure I understand this logic...is this trying to determine if a path as a string represents a folder?  It feels like it should have some tests on it to make it clear it works